src/toggle-switch/toggle-switch.scss

```diff
@@ -2,194 +2,173 @@
-      border-radius: $iui-baseline;
-      @media (prefers-reduced-motion: no-preference) {
-        transition: background-color $iui-speed-fast ease, border-color $iui-speed-fast ease;
-      }
-      @include themed {
-        background-color: t(iui-color-background-1);
-        border: 1px solid rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
-      }
-
-      > .iui-icon {
-        opacity: 0;
-        width: $iui-sm;
-        height: $iui-sm;
-        display: inline-block;
-        position: relative;
-        left: $iui-xs;
-        @media (prefers-reduced-motion: no-preference) {
-          transition: opacity $iui-speed-fast ease;
-        }
-        @include themed {
-          fill: t(iui-color-foreground-accessory);
-        }
-      }
-
-      > .iui-handle {
-        position: absolute;
-        height: $iui-m;
-        width: $iui-m;
-        top: 50%;
-        transform: translateY(-50%);
-        right: $iui-m + $iui-xs;
-        border-radius: 50%;
-        @media (prefers-reduced-motion: no-preference) {
-          transition: right $iui-speed-fast ease, background-color $iui-speed-fast ease, opacity $iui-speed-fast ease;
-        }
-        @include themed {
-          background-color: t(iui-color-foreground-body);
-          opacity: t(iui-opacity-2);
-        }
-      }
+  > * + * {
+    margin-left: $iui-s;
+    @supports (gap: $iui-s) {
+      margin-left: 0;
     }
+  }
 
-    &:hover:not(:disabled),
-    &:focus {
-      ~ .iui-toggle {
-        @include themed {
-          border: 1px solid rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-2));
-        }
-
-        > .iui-handle {
-          @include themed {
-            opacity: t(iui-opacity-1);
-          }
-        }
-      }
+  // Disabled
+  &.iui-disabled {
+    cursor: not-allowed;
+    @include themed {
+      color: t(iui-text-color-muted);
     }
+  }
+
+  // For labels on the left
+  &.iui-label-on-left {
+    grid-template-areas: 'label toggle';
+  }
+
+  // For labels on the right
+  &.iui-label-on-right {
+    grid-template-areas: 'toggle label';
+  }
+}
+
+@mixin iui-toggle-switch {
+  $iui-toggle-switch-handle-size: $iui-icons-default;
+  $iui-toggle-switch-handle-offset: $iui-component-offset;
+  $iui-toggle-switch-border-thickness: 1px;
+
+  @include iui-reset;
+  @include iui-focus($thickness: $iui-toggle-switch-border-thickness, $offset: $iui-toggle-switch-border-thickness);
+  grid-area: toggle;
+  display: flex;
+  position: relative;
+  cursor: pointer;
+
+  // #region IE solution
+  width: $iui-toggle-switch-handle-size;
+  height: $iui-toggle-switch-handle-size;
+  @supports (appearance: none) {
+    appearance: none;
+    width: ($iui-toggle-switch-handle-size + ($iui-toggle-switch-handle-offset * 2)) * 2;
+    height: auto;
+    border-radius: $iui-border-radius-round;
+    box-sizing: border-box;
+    transition: background-color $iui-speed-fast ease-out, border-color $iui-speed-fast ease-out;
```

Comment:
Some transitions in this file aren't surrounded by a `@media( prefers-reduced-motion...)`. I tested them out and they're very subtle. I noticed that in one section there's a subtle version and a more overt version. So I'm assuming this was done on purpose and it's okay? Generally there's no transitions except those in the `@media` block, so just making sure.

Reply:
The `reduced-motion` guard only applies to transitions that consist of some motion. When there is only a color change, it is ok to keep those transitions outside the media query.